### ISSUE TITLE
Agent makes HTTP connection to adapter

### ISSUE THREAD

**Issue by bcampbell-ts** [mtconnect/mtconnect-agent#46]
The `connectToDevice` function contains the following:
```javascript
  const response = request(`http://${ip}:${port}`)
  response.pipe(processSHDR(uuid))
```

This is connecting to the adapter using an HTTP client. While this could work, it appears it doesn't since most adapters don't send any HTTP headers.

The agent generates the following error:
```
{ Error: Parse Error
    at Error (native)
    at Socket.socketOnData (_http_client.js:363:20)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Socket.Readable.push (_stream_readable.js:134:10)
    at TCP.onread (net.js:547:20) bytesParsed: 0, code: 'HPE_INVALID_CONSTANT' }
```
HPE_INVALID_CONSTANT meaning HTTP Parse Error

**Comment by wsobel** [comment]
This is for the first round trip to get the device information from the adapter using HTTP. The second connection is to the adapter SHDR feed. If you are confusing the two, then this is most likely the source of the problem.

**Comment by bcampbell-ts** [comment]
The call to `processSHDR` which I included in the code snippet above is the only call I see in the entire code base. So, I don't see how it is not using the HTTP request.
Additionally, it is connecting to the ip and port which I specified in a Data field inserted into the probe XML response.

**Comment by wsobel** [comment]
Looking through the code, it appears to be using HTTP for the connection. This is incorrect and will need to be fixed.

**Comment by wsobel** [comment]
This has now been addressed in commit: cb1bc4904

It now connects to shdr using a straight line reader and the heartbeat (PING/PONG) protocol is also present. This still needs more work since it should be refactored into a separate module. This is next up.

**Comment by bcampbell-ts** [comment]
Works as expected
